src/screens/SendFunds/03b-AmountNft.js

```diff
@@ -0,0 +1,203 @@
+// @flow
+import React, { useCallback, useState, useEffect } from "react";
+
+import { BigNumber } from "bignumber.js";
+import { useSelector } from "react-redux";
+import { useTranslation } from "react-i18next";
+import { useNavigation, useTheme } from "@react-navigation/native";
+import { getAccountBridge } from "@ledgerhq/live-common/lib/bridge";
+import { View, StyleSheet, SafeAreaView, TextInput } from "react-native";
+import useBridgeTransaction from "@ledgerhq/live-common/lib/bridge/useBridgeTransaction";
+
+import type { Transaction } from "@ledgerhq/live-common/lib/types";
+
+import { accountScreenSelector } from "../../reducers/accounts";
+import TranslatedError from "../../components/TranslatedError";
+import KeyboardView from "../../components/KeyboardView";
+import Button from "../../components/Button";
+import LText from "../../components/LText";
+import { ScreenName } from "../../const";
+
+const forceInset = { bottom: "always" };
+
+type Props = {
+  navigation: any,
+  route: {
+    params: {
+      accountId: string,
+      transaction: Transaction,
+    },
+  },
+};
+
+export default function SendAmountNFT({ route }: Props) {
+  const navigation = useNavigation();
+  const { t } = useTranslation();
+
+  const { colors } = useTheme();
+  const { account, parentAccount } = useSelector(accountScreenSelector(route));
+  const [quantity, setQuantity] = useState(0);
+  const [error, setError] = useState({});
+
+  const {
+    transaction,
+    setTransaction,
+    status,
+    bridgePending,
+    // bridgeError,
+  } = useBridgeTransaction(() => ({
+    transaction: route.params.transaction,
+    account,
+    parentAccount,
+  }));
+
+  useEffect(() => {
+    const bridge = getAccountBridge(account, parentAccount);
+    setTransaction(
+      bridge.updateTransaction(transaction, {
+        quantities: [new BigNumber(quantity ?? 0)],
```

Comment:
⚠️  this code runs on `mount` and makes the quantity go back to 0 even when it wasn't set.

the way to fix this is to not use an effect but instead to use a callback that effectively do this and instead of having a local quantity state, you directly get it from transaction.quantities..